openshift/machine-config-operator
# ovirt: Use the DNSVIP for workers

oVirt worker will most of the time failed to join the cluster because they don't have a nameserver
set to resolve the api or api-int.
The problem is that a `/etc/dhclient/dhclient.conf` is written during the network manager pre-up hook, using the file https://github.com/openshift/machine-config-operator/commit/d3d4e5fd662bd54c04cd38374d3cc28ceb7c4dff#diff-35faebd7db8179e62b9eb7ca9604ce72
and because of a race, dhclient doesn't recognize the new config.

- What I did
This change will write `/etc/dhcp/dhclient.conf` using ignition, with the DNS VIP prepended 
as the first nameserver. The DNSVIP is interpolated by the server so it is written in its final form
in the ignition phase (just like the masters). This eliminates the race I'm seeing.

Related-to: https://github.com/openshift/machine-config-operator/issues/1312


- How to verify it

Run IPI installation of oVirt, 3 master (all of them use those static pods) and 2 workers which are added using cluster-api-provider-ovirt, and see them join and go into `Ready` state.

- Description for the changelog
Set the `DNSVIP` as the first nameserver for worker by writing `/etc/dhcp/dhclient.conf` using ignition.
